### PR TITLE
Add crowdsourcing example

### PR DESCRIPTION
**Pull request by henryre**
Add a crowdsourcing example based on https://data.world/crowdflower/weather-sentiment. We provide high quality crowd labels for half of the training and dev set, and use text-based LFs to cover the rest. We then train a final model over the tweets themselves. The data set is small, and even our MLP over BoW is unstable. However, it's effective for demonstration purposes.

**Test plan**
Travis runs

### CODE REVIEW

File: crowdsourcing/crowdsourcing_tutorial.py [hunk 1]
```diff
@@ -19,161 +19,214 @@
 # * Tweet not related to weather condition
 #
 # The catch is that 20 contributors graded each tweet, and in many cases contributors assigned
-# conflicting sentiment labels to the same tweet.
+# conflicting sentiment labels to the same tweet. Our goal is to label each tweet as either
+# positive or negative so that we can train a language model over the tweets themselves.
+#
 # This is a common issue when dealing with crowdsourced labeling workloads.
```
Comment (brahmaneya) [inline crowdsourcing/crowdsourcing_tutorial.py]: This line was meant to follow "contributors assigned conflicting labels...". You can delete or move it.

File: crowdsourcing/crowdsourcing_tutorial.py [hunk 2]
```diff
@@ -19,161 +19,214 @@
...
+# We've also altered the data set to reflect a realistic crowdsourcing pipeline
+# where only a subset of our full training set have recieved crowd labels.
+# We'll encode the crowd labels themselves as labeling functions in order to learn trust
+# weights for each crowdworker, and write a few heuristic labeling functions to cover the
```
Comment (brahmaneya) [inline crowdsourcing/crowdsourcing_tutorial.py]: crowdworker -> crowd worker (for consistency)

File: crowdsourcing/crowdsourcing_tutorial.py [hunk 3]
```diff
@@ -19,161 +19,214 @@
...
-labels_by_annotator = crowd_answers[["tweet_id", "worker_id", "emotion"]].groupby(
-    ["worker_id"]
-)
+labels_by_annotator = crowd_answers.groupby("worker_id")
 worker_dicts = {}
 for worker_id in labels_by_annotator.groups:
-    worker_df = labels_by_annotator.get_group(worker_id)[["tweet_id", "emotion"]]
-    worker_dicts[worker_id] = dict(zip(worker_df.tweet_id, worker_df.emotion))
+    worker_df = labels_by_annotator.get_group(worker_id)[["answer"]]
+    v = set(worker_df.answer.tolist())
```
Comment (brahmaneya) [inline crowdsourcing/crowdsourcing_tutorial.py]: v seems to be unused

File: crowdsourcing/crowdsourcing_tutorial.py [hunk 4]
```diff
@@ -19,161 +19,214 @@
...
+    return 0 if label == 0 else -1
+
+
+def get_worker_labeling_function(worker_id, f):
     worker_dict = worker_dicts[worker_id]
+    name = f"worker_{worker_id}"
+    return LabelingFunction(name, f=f, resources={"worker_dict": worker_dict})
+
+
+worker_lfs_pos = [
```
Comment (brahmaneya) [inline crowdsourcing/crowdsourcing_tutorial.py]: (Feel free to ignore) Until we have dependency learning, this might work worse than having a single lf, because the exclusive relationship between the positive and negative lfs for each worker is not encoded anywhere.

File: crowdsourcing/crowdsourcing_tutorial.py [hunk 5]
```diff
@@ -187,49 +240,28 @@ def low_absolute_polarity(x):
...
-)
-
-# %% [markdown]
-# ### Comparison to baseline
-# Finally, we can compare this to a model trained only on the gold labels in the dev set. This shows us the benefit of u
+model = tf.keras.Sequential()
+model.add(tf.keras.layers.Dense(10, activation=tf.nn.relu))
+model.add(tf.keras.layers.Dense(2, activation=tf.nn.softmax))
+model.compile("Adam", "categorical_crossentropy")
+callbacks = model.fit(X_train, Y_train_prob, epochs=10, verbose=0)
```
Comment (brahmaneya) [inline crowdsourcing/crowdsourcing_tutorial.py]: callbacks is unused. Maybe _ = ?

File: crowdsourcing/crowdsourcing_tutorial.py (hunk 4 above)
Comment (henryre) [reply crowdsourcing/crowdsourcing_tutorial.py]: Good call. Let's leave for now (for simplicity), but make a note to revisit.
